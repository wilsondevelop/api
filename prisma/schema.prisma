generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Member {
  id         Int         @id @default(autoincrement())
  name       String      @unique
  createdAt  DateTime?   @default(now())
  exercicies Exercicie[]

  @@map("members")
}

model Exercicie {
  id          Int        @id @default(autoincrement())
  memberId    Int
  name        String     @unique
  video       String
  description String?
  createdAt   DateTime?  @default(now())
  member      Member     @relation(fields: [memberId], references: [id])
  trainings   Training[]

  @@map("exercicies")
}

model Training {
  id          Int       @id @default(autoincrement())
  exercicieId Int
  personalId  String
  customerId  String
  video       String
  description String?
  createdAt   DateTime? @default(now())
  expiresIn   DateTime?
  repeat      String
  serie       String
  exercicie   Exercicie @relation(fields: [exercicieId], references: [id])

  @@map("trainings")
}

model TrainingLog {
  id         Int       @id @default(autoincrement())
  trainingId Int
  customerId String
  isDone     Boolean?  @default(true)
  createdAt  DateTime? @default(now())

  @@map("log_trainings")
}
