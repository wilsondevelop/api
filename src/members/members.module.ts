import { Module } from '@nestjs/common';
import { MembersService } from './members.service';
import { MembersController } from './members.controller';
import { PrismaService } from '../shared/database/prisma/prisma.service';

@Module({
  controllers: [MembersController],
  providers: [MembersService, PrismaService]
})
export class MembersModule {}
