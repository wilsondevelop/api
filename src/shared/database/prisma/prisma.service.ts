import {
  Injectable,
  OnModuleInit,
  INestApplication,
  OnModuleDestroy,
} from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class PrismaService
  extends PrismaClient
  implements OnModuleInit, OnModuleDestroy
{
  constructor() {
    super();
  }

  async onModuleInit() {
    await this.$connect();
  }

  async onModuleDestroy() {
    await this.$disconnect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.$on('beforeExit', async () => {
      await app.close();
    });
  }

  async cleanDatabase() {
    if (process.env.NODE_ENV === 'production') return;
    const models = Reflect.ownKeys(this).filter((key) => key[0] !== '_');
    return Promise.all(models.map((modelKey) => this[modelKey].deleteMany()));
  }
}
