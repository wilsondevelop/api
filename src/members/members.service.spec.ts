import { Test, TestingModule } from '@nestjs/testing';
import { PrismaService } from '../shared/database/prisma/prisma.service';
import { MembersService } from './members.service';
import { ConfigService } from '@nestjs/config';

describe('MembersService', () => {
  let service: MembersService;
  let prismaService: PrismaService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [MembersService, PrismaService, ConfigService],
    }).compile();

    prismaService = module.get(PrismaService)

    service = module.get<MembersService>(MembersService);
    await prismaService.member.deleteMany()
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });

  describe('List all members', () => {
    it('should return array members', async () => {
      const findAllSyOn = jest.spyOn(service, 'findAll');
      const members = await service.findAll();
      expect(findAllSyOn).toBeCalled();
      expect(Array.isArray(members)).toBe(true);
    });
  });

  describe('Create member', () => {
    it('should return object member', async () => {
      const createSpyOn = jest.spyOn(service, 'create');
      const member = await service.create({ name: 'member_any' });
      expect(createSpyOn).toBeCalled();
      expect(member).toMatchSnapshot()
    });
  });

    describe('FindOne member', () => {
      it('should return  member', async () => {
        const findSpyOn = jest.spyOn(service, 'findOne');
        const member = await service.findOne(27);
        expect(findSpyOn).toBeCalled();
        expect(member).toMatchSnapshot();
      });
    });
});
