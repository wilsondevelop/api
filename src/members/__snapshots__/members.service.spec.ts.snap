// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersService Create member should return object member 1`] = `
Object {
  "createdAt": 2022-12-03T21:07:18.654Z,
  "id": 30,
  "name": "member_any",
}
`;

exports[`MembersService FindOne member should return  member 1`] = `null`;
